core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -105,6 +109,16 @@ object DumpLogSegments {
         })
       }
     }
+
+    misMatchesForTimeIndexFilesMap.foreach {
+      case (fileName, listOfMismatches) => {
+        System.err.println("Mismatches in :" + fileName)
+        listOfMismatches.foreach(m => {
+          System.err.println("  Index timestamp: %d, log timestamp: %d".format(m._1, m._2))
```

Comment:
We will need to explain the output better. Perhaps when we print "Mismatches in :" , we can say these are cases where the timestamp in the wrapper message doesn't match the inner. Also, it would also be useful to capture errors in which the timestamps are not increasing in the index.